router/voteCallbacks.go

```diff
@@ -73,6 +74,37 @@ func (r *Router) submitEnvelope(request routerRequest) {
 	}
 }
 
+func (r *Router) getStats(request routerRequest) {
+	stats := new(models.VochainStats)
+
+	stats.BlockHeight = uint32(r.vocapp.State.Header(true).Height)
+	stats.BlockTimeStamp = int32(r.vocapp.State.Header(true).Timestamp)
+
+	stats.EntityCount = r.Scrutinizer.EntityCount()
```

Comment:
We should cache the entity count as we do with `countTotalEnvelopes` (see scrutinizer.go line 65)

Reply:
Currently the scrutinizer registers entities in `newEmptyProcess`, where it uses the db `Upsert` method: 

`"Upsert inserts the record into the badgerhold if it doesn't exist. If it does already exist, then it updates the existing record"`

In order to cache the entity height, afaik, here we would need to first check if the entity exists already, which means an extra database query on every new process. Is this trade-off worth it @p4u ?

Reply:
The Upsert() of newEmptyProcess seems not correct to me, since the CreationTime is being updated on each new process. So we should change it and make two operations (see if exists and create it if not). Then we can keep an Entity count cache as we do with countTotalEnvelopes. If think is worth it, we need to minimize the DB transactions in order to scale.

Reply:
This makes sense to me, I'll update the code